core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/Utils.kt

```diff
@@ -28,3 +30,45 @@ fun <T : DataFrame<*>> T.alsoDebug(println: String? = null, rowsLimit: Int = 20)
         print(borders = true, title = true, columnTypes = true, valueLimit = -1, rowsLimit = rowsLimit)
         schema().print()
     }
+
+fun DataFrame<*>.toCode(variableName: String = "df"): String =
+    buildString {
+        append("val $variableName = dataFrameOf(\n")
+        appendColumns(this@toCode, indent = 1)
+        append(")")
+    }
+
+private fun StringBuilder.appendColumns(df: DataFrame<*>, indent: Int) {
+    val pad = "    ".repeat(indent)
+    df.getColumns { colsAtAnyDepth().simplify() }.forEach { col ->
+        if (col is ColumnGroup<*>) {
+            append("$pad\"${col.name()}\" to columnOf(\n")
+            appendColumns(col, indent + 1)
+            append("$pad),\n")
+        } else {
+            appendColumn(col, pad)
+        }
+    }
+}
+
+private fun StringBuilder.appendColumn(column: DataColumn<Any?>, pad: String) {
+    append("$pad\"${column.name()}\" to columnOf(")
+    append(column.values().joinToString(", ") { it.toLiteral() })
+    append("),\n")
+}
+
+private fun Any?.toLiteral(): String =
```

Comment:
You can supply a KType to the function, which saves instance checks :)

Reply:
How free? :o Maybe i don't get the idea

Reply:
let's keep instance checks, i added a lot of branches where smart casts are useful